src/Compilers/CSharp/Test/Semantic/Semantics/DeconstructionTests.cs

```diff
@@ -2620,5 +2634,91 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_TypeVarNotFound, "var").WithLocation(6, 15)
             );
         }
+
+        [Fact]
+        public void MixedDeconstruction_01()
+        {
+            string source = @"
+class Program
+{
+    static void Main(string[] args)
+    {
+        var t = (1, 2);
+        var x = (int y, int z) = t;
+    }
+}";
+
+            var comp = CreateCompilationWithMscorlib(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef });
+            comp.VerifyDiagnostics(
+                // (7,18): error CS8185: A declaration is not allowed in this context.
```

Comment:
Because per LDM we need a separate decision (from Jared) when  (i.e. either C# 7 or 7.1 or 8) to permit the declarations in deconstructions in contexts other than an expression statement containing the deconstruction assignment and an expression among the for-loop initializers. I believe he is likely to make that decision tomorrow.

---
In reply to: [90169794](https://github.com/dotnet/roslyn/pull/15548#discussion_r90169794) [](ancestors = 90169794)